src/ui/replay_parser.rs

```diff
@@ -2366,6 +2367,192 @@ impl ToolkitTabViewer<'_> {
+                if let Some(v) = damage_report.flooding { damage_dealt.insert("damage_flooding".to_string(), json!(v)); }
+            }
+            
+            let damage_dealt = serde_json::Value::Object(damage_dealt);
+
+            // Build potential damage section
+            let potential_damage = serde_json::json!({
+                "potential_damage_total": potential_damage_total,
+                "potential_damage_artillery": report.potential_damage_report.as_ref().map(|d| d.artillery),
+                "potential_damage_torpedoes": report.potential_damage_report.as_ref().map(|d| d.torpedoes),
+                "potential_damage_planes": report.potential_damage_report.as_ref().map(|d| d.planes)
+            });
+
+            // Build damage received section using structured data
+            let results_info = report.vehicle.results_info().and_then(|info| info.as_array());
+            let constants = report.vehicle.player().map(|_| ui_report.constants.read());
+
+            let get_received_damage = |damage_type: &str| {
+                if let (Some(info_array), Some(constants)) = (results_info, constants.as_ref()) {
+                    constants.pointer(format!("/CLIENT_PUBLIC_RESULTS_INDICES/received_{}", damage_type).as_str())
+                        .and_then(|idx| idx.as_u64())
+                        .and_then(|idx| info_array[idx as usize].as_number())
+                        .and_then(|num| num.as_u64())
+                } else {
+                    None
+                }
+            };
+
+            let damage_received = serde_json::json!({
+                "damage_received_total": damage_received_total,
+                "damage_received_ap": get_received_damage(DAMAGE_MAIN_AP),
+                "damage_received_he": get_received_damage(DAMAGE_MAIN_HE),
+                "damage_received_sap": get_received_damage(DAMAGE_MAIN_CS),
+                "damage_received_he_secondary": get_received_damage(DAMAGE_ATBA_HE),
+                "damage_received_sap_secondary": get_received_damage(DAMAGE_ATBA_CS),
+                "damage_received_torpedoes": get_received_damage(DAMAGE_TPD_NORMAL),
+                "damage_received_deep_water_torpedoes": get_received_damage(DAMAGE_TPD_DEEP),
+                "damage_received_fire": get_received_damage(DAMAGE_FIRE),
+                "damage_received_flooding": get_received_damage(DAMAGE_FLOOD)
+            });
+
+            // Build experience section
+            let experience = serde_json::json!({
+                "base_xp": report.base_xp,
+                "raw_xp": report.raw_xp
+            });
+
+            // Build battle life section
+            let battle_life = serde_json::json!({
+                "time_lived_secs": report.time_lived_secs,
+                "distance_traveled": report.distance_traveled
+            });
+
+            // Build combat achievements section
+            let combat = serde_json::json!({
```

Comment:
I think the raw results data also records _actual_ achievements. TODO on my end:

- Add the actual battle achievements
- Add kills

Not necessary for the baseline data export feature, just a note for myself.